apps/server/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { AuthService } from './auth.service';
+
+import { Request as HttpRequest } from 'express';
+
+import { AuthenticationGuard } from './guards/google-oauth-guard';
+
+import { Controller, Get, Request, UseGuards } from '@nestjs/common';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get()
+  @UseGuards(AuthenticationGuard)
+  async googleAuth(@Request() req: HttpRequest) {}
+
+  @Get('redirect')
+  @UseGuards(AuthenticationGuard)
+  authenticationRedirect(@Request() req: HttpRequest) {
+    if (req.user) {
+    }
+    return this.authService.requestAuthentication(req);
```

Comment:
Can you explain what this piece of code is doing? @AsminShresthaLF

Reply:
Requests authentication to check if request originates from vyaguta/some interface if request has  x-api-key as the `STATIC_API_TOKEN` and is currently just an implementation for such scenario.

Reply:
What about this block of code?
```tsx
if (req.user) {
}
```

Reply:
Doesn't exist in new PR.